### PR TITLE
Codeql analysis

### PR DESCRIPTION
**Pull request by mvr320**
Currently it only finds issues with upstream repos for javascript so this is on the very safe side.

I'm not 100% if the invocation of build-scripts is correct here as I don't know where those should be stored?

I think @meisterT @eldering @thijskh all have an opinion on how such a tool should act so choose them as reviewers.

### CODE REVIEW

**Comment by meisterT** [comment]
Where can I see a result of this?

**Comment by mvr320** [comment]
> 
> 
> Where can I see a result of this?

Currently it does not have security items, but in the future you can see this under the security tab in GitHub. As I said its currently very on the safe side. The picture here is from an earlier run where I didn't remove the upstream code yet:
image

But it will basicly fail in case someone enters a new bug again (which is non-blocking but should be looked at)

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        ]
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'java', 'javascript', 'python', 'ruby' ]
```
Comment (eldering) [inline .github/workflows/codeql-analysis.yml:29]: Not sure it makes sense to include ruby here, as we have no ruby code (except the test sources).

**Review by eldering — APPROVED** [review]
Looks fine to me if this indeed only shows up in a separate security tab that doesn't clutter the PR.

**Pull request merged**